include/pybind11/detail/type_caster_base.h

```diff
@@ -469,40 +469,30 @@ PYBIND11_NOINLINE bool isinstance_generic(handle obj, const std::type_info &tp)
     return isinstance(obj, type);
 }
 
-PYBIND11_NOINLINE std::string error_string() {
-    if (!PyErr_Occurred()) {
+PYBIND11_NOINLINE std::string error_string(PyObject *type, PyObject *value, PyObject *trace) {
```

Comment:
OK as-is, but `err_type` or `exc_type` (and `err_value`, `err_trace`) are strongly preferred, to make this code more self-explanatory.